README.md

```diff
@@ -3,12 +3,22 @@
 [![GitHub Super-Linter](https://github.com/interTwin-eu/T6.5-AI-and-ML/actions/workflows/lint.yml/badge.svg)](https://github.com/marketplace/actions/super-linter)
 [![GitHub Super-Linter](https://github.com/interTwin-eu/T6.5-AI-and-ML/actions/workflows/check-links.yml/badge.svg)](https://github.com/marketplace/actions/markdown-link-check)
 
-The entry point of a workflow is given by the orchestrator script `run-workflow.py` .
+## Running a workflow in CWL
+
+The entry point of a workflow is given by the orchestrator script `workflow.cwl`.
+You can run it with any tool that supports the CWL worklfow format, e.g. `cwltool`.
 
 See some examples of workflow executions in `examples.sh` , for instance:
 
 ```bash
-conda run -p ./.venv python run-workflow.py -f ./use-cases/mnist/training-workflow.yml
```

Comment:
I would still leave `run-workflow.py` as the entry point, and run cwltool from within, to abstract from the actual CWL engine. Furthermore, `run-workflow.py` could still automatize the deployment of workflow steps before the workflow is executed. Alternatively, another solution would be to deploy each step of a workflow (i.e., python envs) as first step of CWL workflow. Try both and choose which you prefer

Reply:
I tried out setting up the environment with the `cwltool`, but it's a bit quirky, since CWL expects environments deployed in containers and not conda environments, so I invoke `cwltool` from `run-workflow.py` as you suggested.
However, the CWL format requires a specific way how to write the working flow definition file `training-workflow-cwl.yml`, so there is a redundancy in path definitions in the setup and execution stage, since your style of setting up the environments is different from CWL style. This makes it a bit unelegant, but this should resolve itself later, since we'll have the environment setup in containers later.